### PR TITLE
Configure the refbox through frontend

### PR DESCRIPTION
**Pull request by TarikViehmann**
This PR adds the functionalities required to configure and reconfigure the refbox through the frontend (see https://github.com/robocup-logistics/rcll-refbox-frontend/pull/59).
In oder to make this possible, a synchronization between the yaml-cpp configuration files, the refbox confval facts and the refbox globals is established.

There are three differnt parts to this.

## Updates of yaml config values and Globals
This one is straght forwards, configuration values and respective clips knowledge is updated according to the changes.
When changing a value, the type must be preserved, otherwise the update is rejected (e.g., you cannot change the exploration time from 180 to "this is a string").
However, changing arbitrary config values at arbitrary times is likely to not yield the expected results as the refbox goes through different stages and will not automatically revisit every decision made so far based on a config change.
For example, changing the field size after the field was already generated will do nothing.

Hence, best results can be achieved by configuring the refbox before starting the setup phase.

Once feature I did test and that works well is to switch the connection mode of machines during a game.
E.g., if a machine is having hardware issues mid-game, setting the connection mode to `mockup` will allow the game to proceed. It is even possible to later switch it back to `plc` and to re-introduce a machine again through that.
Also here, a small caveat: This will only work if the machine in question is IDLE or READY-AT-OUTPUT, as during processing there would be unknown states.

## Loading of Presets
Analogously to the command-line options that allow to load different yaml files, these files can also be loaded at run-time.
Similar restrictions as the ones above for individual config values apply. Additionally, it is not possible to directly reverse such an operation (unloading a file after it has been loaded). Most notably, loading a challenge preset will change the bahavior or the refbox, so if you do that by accident, you would need to reset the refbox

## Reset the Refbox
Restart the refbox completely. The refbox will relaunch with the same command-line arguments again.
This behavior is also available from outside of the frontend, by just sending SIGUSR1 signal to the process: `kill -SIGUSR1 $(pgrep llsf-refbox)`
In order to make this happen, it is necessary that all components shut down gracefully, which can be kinda tricky in this heavily multi-threaded environment.
While I tested this for basic scenarios, it could very well be that this is not completely stable yet.

One of the components that caused issues was the rest-api/webview which was non-functional anyways. Hence this PR removes it.

### CODE REVIEW

File: src/games/rcll/challenges.clp
Comment (snoato) [inline src/games/rcll/challenges.clp:437]: are these rules necessary given that the global is initialised from the config value anyways?

File: src/games/rcll/config.clp
Comment (snoato) [inline src/games/rcll/config.clp:83]: why do we have two values?  this seems dangerous in terms of usability

File: src/games/rcll/time.clp
Comment (snoato) [inline src/games/rcll/time.clp:16]: Why the -_ mixing?

File: src/games/rcll/utils.clp
Comment (snoato) [inline src/games/rcll/utils.clp:501]: shouldn't we move this to a config file too and do a lookup from there?

**Review by snoato — DISMISSED** [review]
Great addition! Got some remarks and questions before approval.

File: src/games/rcll/challenges.clp
Comment (TarikViehmann) [reply src/games/rcll/challenges.clp:437]: yes the global can actually change when the challenge track is loaded

File: src/games/rcll/config.clp
Comment (TarikViehmann) [reply src/games/rcll/config.clp:83]: i agree this design is bad, it was introduced when challenges were first introduced.
The idea is that challenge configs have their own prefix to make configuration easier, but in hindsight i regret that as it introduces these weird situations and also we are not consistent with it anyways (grasping challenge needs to also alter the game config to load the field).

But I would consider this out of scope for this PR to also rewrite challenge configs.

File: src/games/rcll/utils.clp
Comment (TarikViehmann) [reply src/games/rcll/utils.clp:501]: we could but it just introduces yet another load of yaml files that pretty much never get touched anyways, right?
We changed the tags once in the last 8 years, hence I think there is no need to make it configurable.

File: src/games/rcll/time.clp
Comment (TarikViehmann) [reply src/games/rcll/time.clp]: I have no idea :) fixed it, thank you!

File: src/games/rcll/challenges.clp
Comment (snoato) [inline src/games/rcll/challenges.clp:116]: is there a reason why we mix upper and lower case (sometimes even within one global)?

**Review by snoato — APPROVED** [review]
One Q, otherwise great work!

**Pull request merged**
